### DISCUSSION TITLE
Directly Switch to Light or Dark Theme

### DISCUSSION THREAD

**Discussion by sf1tzp** [ellisonleao/gruvbox.nvim#297] [Q&A]
Hi! I'm pretty sure this is a nooby question, but... I am new to vim and still learning the ropes.

Since this plugin requires setting `background` independently of `colorscheme gruvbox` there doesn't seem to be a way to 'directly' switch between the two settings. 

I'm using Telescope to switch colorschemes, and there's only one `gruvbox` entry. If I'm currently using a light theme and want to switch to gruvbox with a dark background, I need to first switch to gruvbox and then change the background setting. 

I noticed [some themes](https://github.com/kepano/flexoki-neovim/tree/main/colors) ship with multiple themes, but I'm assuming that since the change is so minimal, you didn't feel the need to do that here. 

`:help colorscheme` says we can drop things in  `~/.config/nvim/colors`, so I tried a few things there:
``` lua
vim.o.background = "light"
vim.cmd([[colorscheme gruvbox]])
```
and
``` vim
runtime gruvbox
set background=dark
```
I also tried adding gruvbox twice my lazy config with different `name` and `config` functions, but yeah, no luck.

So I just thought I'd ask. Is there convenient or canonical way of creating `gruvbox-light` and `gruvbox-dark` colorschemes to switch to?

**Comment by polyzen** [comment]
Simply using `:set background=dark` or `:set background=light` should work.

https://github.com/tpope/vim-unimpaired comes with mappings such as `yob` which toggles the `background` option.

https://github.com/vimpostor/vim-lumen could be used to automatically switch based on your system dark/light mode.

**Comment by sf1tzp** [comment]
Thank you for the suggestions,

Using `:set background` does work but it isn't exactly the functionality that I'm looking for here. I want to "switch to gruvbox and set the background to dark" using a `:colorscheme` command.

**Comment by sf1tzp** [comment]
I added a couple of variants to the theme in #298 for what it's worth - still open to suggestions though if I am reinventing the wheel here

**Comment by sf1tzp** [comment]
Using files like this works well if anybody else is trying to achieve this:
```
-- ~/.config/nvim/colors/gruvbox-dark.lua
vim.o.background = "dark"
require("gruvbox").load()
```
```
-- ~/.config/nvim/colors/gruvbox-light.lua
vim.o.background = "light"
require("gruvbox").load()
```
<img width="1417" alt="Screenshot 2023-11-21 at 11 44 38 PM" src="https://github.com/ellisonleao/gruvbox.nvim/assets/84053270/3e23ffc8-2edf-407e-a3ee-bc3e96f19871">
<img width="1402" alt="Screenshot 2023-11-21 at 11 50 36 PM" src="https://github.com/ellisonleao/gruvbox.nvim/assets/84053270/d8a2387b-6be7-4dca-937e-7952226ade3f">
